cli/src/pcluster/config/update_policy.py

```diff
@@ -149,7 +151,7 @@ def __eq__(self, other):
     fail_reason=lambda change, patch: "Shrinking a queue requires the compute fleet to be stopped first",
     action_needed=UpdatePolicy.ACTIONS_NEEDED["pcluster_stop"],
     condition_checker=lambda change, patch: not patch.cluster.has_running_capacity()
-    or change.new_value >= change.old_value,
+    or (change.new_value or DEFAULT_MAX_COUNT) >= (change.old_value or DEFAULT_MAX_COUNT),
```

Comment:
The  `or` approach is not the good one when working with numbers, if new_value is 0 the code will use DEFAULT_MAX_COUNT, the same is true for old_value.

```
>>> print(0 or "default_value")
default_value
```

Reply:
let's add unit tests with new_value or old_value set to 0

Reply:
You are right... no issue, we can write `change.new_value if change.new_value is not None else DEFAULT_MAX_COUNT`